src/pocapi.hpp

```diff
@@ -14,5 +14,70 @@
 // limitations under the License.
 //*****************************************************************************
+#include <stddef.h>
+#include <stdint.h>
+
+struct OVMS_Status;
+
+// TODO reuse this in precision.hpp
+enum DataType {
+    OVMS_DATATYPE_BF16,
+    OVMS_DATATYPE_FP64,
+    OVMS_DATATYPE_FP32,
+    OVMS_DATATYPE_FP16,
+    OVMS_DATATYPE_I64,
+    OVMS_DATATYPE_I32,
+    OVMS_DATATYPE_I16,
+    OVMS_DATATYPE_I8,
+    OVMS_DATATYPE_I4,
+    OVMS_DATATYPE_U64,
+    OVMS_DATATYPE_U32,
+    OVMS_DATATYPE_U16,
+    OVMS_DATATYPE_U8,
+    OVMS_DATATYPE_U4,
+    OVMS_DATATYPE_U1,
+    OVMS_DATATYPE_BOOL,
+    OVMS_DATATYPE_CUSTOM,
+    OVMS_DATATYPE_UNDEFINED,
+    OVMS_DATATYPE_DYNAMIC,
+    OVMS_DATATYPE_MIXED,
+    OVMS_DATATYPE_Q78,
+    OVMS_DATATYPE_BIN
+};
+
+enum BufferType {
+    OVMS_BUFFERTYPE_CPU,
+    OVMS_BUFFERTYPE_CPU_PINNED,
+    OVMS_BUFFERTYPE_GPU,
+    OVMS_BUFFERTYPE_HDDL,
+};
+
+struct OVMS_Status;
+struct OVMS_InferenceRequest;
+struct OVMS_InferenceResponse;
+
+// OVMS_InferenceRequest
+OVMS_Status* OVMS_InferenceRequestNew(char* modelName, uint32_t servableVersion);
+OVMS_Status* OVMS_InferenceRequestDelete(OVMS_InferenceRequest* response);
+OVMS_Status* OVMS_InferenceRequestAddInput(OVMS_InferenceRequest* request, char* inputName, DataType datatype, uint64_t* shape, uint32_t dimCount);
+OVMS_Status* OVMS_InferenceRequestAddInputRaw(OVMS_InferenceRequest* request, char* inputName, DataType datatype);  // TODO consider no datatype & handle the parameters
+// ownership of data needs to be maintained during inference
+OVMS_Status* OVMS_InferenceRequestInputSetData(OVMS_InferenceRequest* request, char* inputName, void* data, size_t bufferSize, BufferType bufferType, uint32_t deviceId);
+OVMS_Status* OVMS_InferenceRequestInputRemoveData(OVMS_InferenceRequest* request, char* inputName);
+OVMS_Status* OVMS_InferenceRequestRemoveInput(OVMS_InferenceRequest* request, char* inputName);  // this will allow for reuse of request but with different input data
+OVMS_Status* OVMS_InferenceRequestRemoveAllInputs(OVMS_InferenceRequest* request);
+OVMS_Status* OVMS_InferenceRequestAddRequestedOutput(OVMS_InferenceRequest* request, char* inputName);  // TODO consider the other way around - add not usefull outputs
+OVMS_Status* OVMS_InferenceRequestAddParameter(OVMS_InferenceRequest* request, char* paramaterName, DataType datatype, void* data, size_t byteSize);
+
+// OVMS_Inference Response
+OVMS_Status* OVMS_InferenceResponseOutputCount(OVMS_InferenceResponse* response, uint32_t* count);
+OVMS_Status* OVMS_InferenceResponseOutput(OVMS_InferenceResponse* response, uint32_t id, char* name, DataType* datatype, uint64_t* shape, uint32_t dimCount, BufferType* bufferType, uint32_t* deviceId, void** data);
+OVMS_Status* OVMS_InferenceResponseDelete(OVMS_InferenceResponse* response);
+OVMS_Status* OVMS_InferenceResponseParameterCount(OVMS_InferenceResponse* response, uint32_t* count);
```

Comment:
```suggestion
OVMS_Status* OVMS_InferenceResponseGetParameterCount(OVMS_InferenceResponse* response, uint32_t* count);
```
What do you think about adding Get for getters?